### PR TITLE
homework3

### PR DESCRIPTION
**Pull request by dmytrotolbatov**

### CODE REVIEW

File: homeworks/dmytro.tolbatov_dmytrotolbatov/homework_3/functionArray.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        result += number;
+    });
+
+    return result;
+}
+
+console.log(squareNumber(4));
+console.log(maxByStep(32, 4));
+console.log(squareArray(maxByStep(32, 4)));
+console.log(sumArray(squareArray(maxByStep(32, 4))));
```
Comment (markoffden) [inline homeworks/dmytro.tolbatov_dmytrotolbatov/homework_3/functionArray.js]: looks good

File: homeworks/dmytro.tolbatov_dmytrotolbatov/homework_3/integerRecursion.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const readline = require('readline');
+
+function integerRecursion(current, maxNumber) {
+    if (current <= maxNumber) {
+        if (current % 2 === 0) {
+            console.log(current);
```
[the remark this answers is not in the document]
Comment (dmytrotolbatov) [reply homeworks/dmytro.tolbatov_dmytrotolbatov/homework_3/integerRecursion.js]: done

File: homeworks/dmytro.tolbatov_dmytrotolbatov/homework_3/integerRecursion.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const readline = require('readline');
+
+function integerRecursion(current, maxNumber) {
+    if (current <= maxNumber) {
+        if (Math.log2(current) % 1 === 0) {
```
Comment (markoffden) [inline homeworks/dmytro.tolbatov_dmytrotolbatov/homework_3/integerRecursion.js]: task is not to use Math methods :)

File: homeworks/misha.kolosov_mwkolosov/homework_2/mirror.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 for (let i = 500; i < 1000; i++) {
     // split().reverse().join();
     const reverse = (i.toString().split('').reverse().join(''));
-    if (i == reverse) {
+    if (i === reverse) {
         console.log(i);
     }
 }
```
Comment (markoffden) [inline homeworks/misha.kolosov_mwkolosov/homework_2/mirror.js]: remove this changes from your commit, work in your directories only pls ;)

File: homeworks/dmytro.tolbatov_dmytrotolbatov/homework_3/integerRecursion.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+const readline = require('readline');
+
+function numberCheck(number) {
+    if (number === 1) {
+        return true;
+    }
+
+    if (number > 0 && Number.isInteger(number)) {
```
Comment (markoffden) [inline homeworks/dmytro.tolbatov_dmytrotolbatov/homework_3/integerRecursion.js]: accepted, but you could simplify it to `if (number > 1)`

**Pull request merged**